tidb/src/tidb/txn.clj

```diff
@@ -68,19 +99,26 @@
           (c/execute! conn [(str "alter table " (table-name i) " cache")])))))
 
   (invoke! [this test op]
-    (let [txn      (:value op)
-          use-txn? (< 1 (count txn))]
-          ;use-txn? false]
-          (if use-txn?
-            (c/with-txn op [c conn {:isolation (util/isolation-level test)
-                                    :before-hook (partial c/rand-init-txn! test conn)}]
-              (assoc op :type :ok, :value
-                     (mapv (partial mop! c test table-count) txn)))
-            (c/with-error-handling op
-              (let [attach-info (if (= :r (-> txn first first)) c/attach-query-info c/attach-txn-info)]
-                (attach-info conn
-                             (assoc op :type :ok, :value
-                                    (mapv (partial mop! conn test table-count) txn))))))))
+    (let [txn (:value op)]
+      (condp = (txn-type test table-count txn)
+
+        :single-stmt-write
+        (c/with-error-handling op (single-stmt-write! conn table-count op))
```

Comment:
Is the `with-error-handling` ensured to handle all the possible results of the `single-stmt-write`, like errors "write conflict" or timeout from which nothing returns?

Reply:
No, it only handle some [known errors](https://github.com/pingcap/jepsen/blob/testing_ci/tidb/src/tidb/sql.clj#L225-L241) and associate them to result records, for those unknowns, it [throws exceptions](https://github.com/pingcap/jepsen/blob/testing_ci/tidb/src/tidb/sql.clj#L243-L246), in which case the txn will be considered as undetermined by jepsen.